### ISSUE TITLE
Call C++ function(exported with embind) from JS don't return result.

### ISSUE THREAD

**Issue by Satancito** [emscripten-core/emscripten#10911]
Using this example
Asyncify [Returning Values](https://emscripten.org/docs/porting/asyncify.html#returning-values)

Calling from javascript GetDigestSize wich was exported using Embind. It runs asyncronous but can't capture the result in a variable. Variable is assigned to 0 but GetDigestSize it continues running, when is completed the value is useless the js function has ended before.

I got the same result with fetch api using this other asyncify example
Asyncify [Making async Web APIs behave as if they were synchronous](https://emscripten.org/docs/porting/asyncify.html#making-async-web-apis-behave-as-if-they-were-synchronous) I'm getting empty string instead of data, fetch is completed later. 
[You can check this issue](https://github.com/emscripten-core/emscripten/issues/10850#issuecomment-611303996)

I need some help.
<hr />

**C++ Code - main.cpp**
```
EM_JS(int, EM_JS_GetDigestSize, (const char* str), {
  // Note how we return the output of handleSleep() here.
  return Asyncify.handleSleep(function(wakeUp) {
    const text = UTF8ToString(str);
    const encoder = new TextEncoder();
    const data = encoder.encode(text);
    out("ask for digest for " + text);
    window.crypto.subtle.digest("SHA-256", data).then(digestValue => {
      out("got digest of length " + digestValue.byteLength);
      // Return the value by sending it to wakeUp(). It will then be returned
      // from handleSleep() on the outside.
      wakeUp(digestValue.byteLength);
    });
  });
});

int GetDigestSize(const String& str)
{
    return EM_JS_GetDigestSize(str.c_str());
}

int main(int argc, char ** argv) 
{
}

EMSCRIPTEN_BINDINGS(my_module) 
{
    using namespace emscripten;
    function("GetDigestSize", &GetDigestSize);
}

```

<hr />

**JS Code needed to instantiate module - InstanceTest.js**

```
window.VModuleTest({ noInitialRun: false }).then(module => {
     window.VMainTest = {
          Module: module
     };
});
```

<hr />

**index.html**

```
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>VModuleTest - WebAssembly</title>
    <link rel="icon" href="Assets/favicon/favicon.ico">
    <script type="text/javascript" src="VModuleTest.js"></script>
    <script type="module" src="InstanceTest.js"></script>
</head>

<body>
    <button onclick="OnBtnClick()">Test</button>
    <script type="text/javascript">
        async function OnBtnClick() {
            var x = window.VMainTest.Module.GetDigestSize("Hello"); 
            console.log("Digest Length: " + x);
        }
    </script>
</body>

</html>
```

<hr />

**Output**

```
Module is initialized
VModuleTest.js:9 ask for digest for Hello
indexTest.html:18 Digest Length: 0
VModuleTest.js:9 got digest of length 32
```
![image](https://user-images.githubusercontent.com/18097869/79175722-18b81280-7dc4-11ea-8a60-aba74613e380.png)

**DESIRED BEHAVIOUR**
All instructions executed in order. Syncronously.
```
Module is initialized
VModuleTest.js:9 ask for digest for Hello
VModuleTest.js:9 got digest of length 32
indexTest.html:18 Digest Length: 32
```

<br /><br /><br /><br /><br /><br /><br />
**EXTRAS**

**Compile command - Compile.ps1**
```
em++.bat `
"mainTest.cpp" `
-o "VModuleTest.js" `
-std=c++17 `
-O2 `
--bind `
-s WASM=1 `
-s SINGLE_FILE=0 `
-s ASYNCIFY=1 `
-s NO_EXIT_RUNTIME=1 `
-s EXTRA_EXPORTED_RUNTIME_METHODS=[`'ccall`',`'cwrap`'] `
-s EXPORT_NAME=`'VModuleTest`' `
-s MODULARIZE=1 `
-s DISABLE_EXCEPTION_CATCHING=0 `
-s ENVIRONMENT=web `
-s ASYNCIFY_IMPORTS=[`'EM_JS_GetDigestSize`'] `
-s EXPORTED_FUNCTIONS=[`'_main`'] `
-s VERBOSE=0 `
```

**em++ version**
![image](https://user-images.githubusercontent.com/18097869/79176372-acd6a980-7dc5-11ea-8e5a-9291cab1522c.png)

**Comment by maierfelix** [comment]
I can approve this issue, same problem with numeric returns.

**Comment by JomaCorpFX** [comment]
A little hard to work (verbose) but it solves the problem, it is possible to jump the bug

https://github.com/emscripten-core/emscripten/issues/10850#issuecomment-611311188

**Comment by Satancito** [comment]
Solved with [#11890](https://github.com/emscripten-core/emscripten/issues/11890#issuecomment-711454589)

Check the code
[EmbindAsync](https://github.com/Satancito/EmbindAsync)
